src/watchdog_fsevents.c

```diff
@@ -634,7 +634,10 @@ watchdog_add_watch(PyObject *self, PyObject *args)
                                           &python_callback, &paths_to_watch));
 
     /* Watch must not already be scheduled. */
-    G_RETURN_NULL_IF(PyDict_Contains(watch_to_stream, watch) == 1);
+    if(PyDict_Contains(watch_to_stream, watch) == 1) {
+        PyErr_Format(PyExc_RuntimeError, "Cannot add watch %S - it is already scheduled", watch);
+        return NULL;
+    }
 
     /* Create an instance of the callback information structure. */
     stream_callback_info_ref = PyMem_New(StreamCallbackInfo, 1);
```

Comment:
This really is an edge case but we should probably also set an exception here.

Reply:
Somehow the comment got attached to the wrong line. It's the next line which I am concerned about, where we check if `PyMem_New` succeeded and return NULL otherwise.

Reply:
Fair point, even though an out of memory situation is likely going to create an issue with creating the exception object as well.